services/QuillLMS/spec/controllers/teachers/progress_reports_controller_spec.rb

```diff
@@ -199,6 +199,76 @@
     end
   end
 
+  describe '#diagnostic_insights' do
+    it 'should render diagnostic_insights' do
+      get :diagnostic_insights
+      expect(response).to render_template 'diagnostic_insights'
+    end
+
+    it 'should assign classrooms to the classrooms that belong to the current_user' do
+      get :diagnostic_insights
+      expect(assigns(:classrooms)).to eq teacher.classrooms_i_teach
+    end
+
+    context 'when the user is a teacher' do
+      context 'who has no email verification record' do
+        it 'should assign needs_email_verification to true' do
+          get :diagnostic_insights
+          expect(assigns(:needs_email_verification)).to be
+        end
+      end
+
+      context 'who has a pending email verification record' do
+        let!(:email_verification) { create(:user_email_verification, user: teacher, verified_at: nil) }
+
+        it 'should assign needs_email_verification to true' do
+          get :diagnostic_insights
+          expect(assigns(:needs_email_verification)).to be
+        end
+      end
+
+      context 'who has a verified email verification record' do
+        let!(:email_verification) { create(:user_email_verification, user: teacher, verified_at: Date.yesterday) }
+
+        it 'should assign needs_email_verification to false' do
+          get :diagnostic_insights
+          expect(assigns(:needs_email_verification)).not_to be
+        end
+      end
+    end
+
+    context 'when the user is an admin' do
```

Comment:
If we do get to clean up the grandfathering in admins for this feature, we get to remove this context block, too, which feels kind of like a win.